apps/src/codebridge/FileBrowser/index.tsx

```diff
@@ -395,81 +312,41 @@ const InnerFileBrowser = React.memo(
         {Object.values(files)
           .filter(f => f.folderId === parentId && shouldShowFile(f))
           .sort((a, b) => a.name.localeCompare(b.name))
-          .map(f => {
-            const {iconName, iconStyle, isBrand} = getFileIconNameAndStyle(f);
-            const iconClassName = isBrand
-              ? classNames('fa-brands', moduleStyles.rowIcon)
-              : moduleStyles.rowIcon;
-            return (
+          .map(f =>
+            !isStartMode && f.type === ProjectFileType.LOCKED_STARTER ? (
+              /* Locked starter files are not draggable, unless we are in start mode */
+              <FileRow
+                key={f.id}
```

Comment:
If you converted this into a function that returns either a file row or a draggable file row, you could avoid repeating all the props by extracting them into a separate constant (enableMenu could be `enableMenu: !dragData?.id || isDraggingLocked` if `isDraggingLocked` became the combination of the two conditions above); again, take or leave.

Reply:
Can you clarify what you mean by this? How would a function prevent having to repeat the props?

Reply:
Proposing a slightly different approach to what I think Emily was suggesting, I'd suggest wrapping up the `Draggable` component into a new Component that's draggable or not.

_I think_ it'd be something along these lines:

```
const WrapperComponent = !isStartMode && f.type === ProjectFileType.LOCKED_STARTER
  ? ({children}) => <>{children}</> // I actually don't know if this needs the <> wrapper or not
  : Draggable

return (
  <WrapperComponent
    data={{id: f.id, type: DragType.FILE, parentId: f.folderId}}
    key={f.id}
    Component="li">
      <FileRow
        file={f}
        isReadOnly={isReadOnly}
        appName={appName}
        hasValidationFile={hasValidationFile}
        isStartMode={isStartMode}
        enableMenu={!dragData?.id}
        setFileType={setFileType}
        handleDeleteFile={handleDeleteFile}
        renameFilePrompt={renameFilePrompt}
      />
  </WrapperComponent>
);
```

Obviously, with whatever extra errata that `TypeScript` needs to be happy and maybe a different name. But the concept is just that we're creating a dynamic wrapper component - if it hits the conditions for being the `Draggable` component, then we should use that. If it _doesn't_, then we swap in some other component that does nothing but spit out the children.

Reply:
Okay, I'm not as fancy as Jim; my thought was something like this...
```
.map(f => {
const isLockedStarter = !isStartMode && f.type === ProjectFileType.LOCKED_STARTER;

    // Common props for FileRow
    const fileRowProps = {
      key: f.id,
      file: f,
      isReadOnly,
      appName,
      hasValidationFile,
      isStartMode,
      setFileType,
      handleDeleteFile,
      renameFilePrompt,
      enableMenu: !dragData?.id || isLockedStarter, // Only enable menu if not dragging or it's locked starter
    };

    return isLockedStarter ? (
      <FileRow {...fileRowProps} />
    ) : (
      <Draggable
        data={{ id: f.id, type: DragType.FILE, parentId: f.folderId }}
        key={f.id}
        Component="li"
      >
        <FileRow {...fileRowProps} />
      </Draggable>
    );
});
```

Reply:
I went with Emily's solution because it was a smoother path with typescript, thanks both!